src/resnet.jl

```diff
@@ -44,83 +44,111 @@ Create a skip projection
+           channel_config, block_config, nclasses = 1000)
 
 Create a ResNet model
 ([reference](https://arxiv.org/abs/1512.03385v1)).
 
 # Arguments
 - `block`: a function with input `(inplanes, outplanes, downsample=false)` that returns
            a new residual block (see [`Metalhead.basicblock`](#) and [`Metalhead.bottleneck`](#))
-- `shortcut_config`: the type of shortcut style (either `:A`, `:B`, or `:C`)
+- `residuals`: a 2-tuple of functions with input `(inplanes, outplanes, downsample=false)`,
+               each of which will return a function that will be used as a new "skip" path to match a residual block.
+              [`Metalhead.skip_identity`](#) and [`Metalhead.skip_projection`](#) can be used here. 
+- `connection`: the binary function applied to the output of residual and skip paths in a block
 - `channel_config`: the growth rate of the output feature maps within a residual block
 - `block_config`: a list of the number of residual blocks at each stage
 - `nclasses`: the number of output classes
 """
-function resnet(; block, shortcut_config, channel_config, block_config, nclasses = 1000)
+function resnet(block, residuals::NTuple{2, Any}, connection = (x, y) -> @. relu(x) + relu(y);
+                channel_config, block_config, nclasses = 1000)
   inplanes = 64
   baseplanes = 64
   layers = []
   append!(layers, conv_bn((7, 7), 3, inplanes; stride = 2, pad = (3, 3)))
   push!(layers, MaxPool((3, 3), stride = (2, 2), pad = (1, 1)))
   for (i, nrepeats) in enumerate(block_config)
+    # output planes within a block
     outplanes = baseplanes .* channel_config
-    if shortcut_config == :A
-      push!(layers, Parallel(+, block(inplanes, outplanes, i != 1),
-                                skip_identity(inplanes, outplanes)))
-    elseif shortcut_config == :B || shortcut_config == :C
-      push!(layers, Parallel(+, block(inplanes, outplanes, i != 1),
-                                skip_projection(inplanes, outplanes[end], i != 1)))
-    end
+    # push first skip connection on using first residual
+    # downsample the residual path if this is the first repetition of a block
+    push!(layers, Parallel(connection, block(inplanes, outplanes, i != 1),
+                                       residuals[1](inplanes, outplanes[end], i != 1)))
+    # push remaining skip connections on using second residual
     inplanes = outplanes[end]
     for _ in 2:nrepeats
-      if shortcut_config == :A || shortcut_config == :B
-        push!(layers, Parallel(+, block(inplanes, outplanes, false),
-                                  skip_identity(inplanes, outplanes[end])))
-      elseif shortcut_config == :C
-        push!(layers, Parallel(+, block(inplanes, outplanes, false),
-                                  skip_projection(inplanes, outplanes, false)))
-      end
+      push!(layers, Parallel(connection, block(inplanes, outplanes, false),
+                                         residuals[1](inplanes, outplanes[end], false)))
```

Comment:
Should this be `residuals[2]`, according to the comment?

Reply:
Yes! Thank you for catching this. Would you like to submit a PR to fix this? If not, I will later today.

Reply:
Ah, sure. https://github.com/FluxML/Metalhead.jl/pull/97